File: core/java/src/main/java/hara/lang/protocol/IClose.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

/** Explicitly closes a resource or traversal cursor. */
@HaraProtocolBinding(namespace = "std.protocol.iclose", name = "IClose")
public interface IClose extends AutoCloseable {
  @Override
  @HaraMethod(value = "close", arity = 1)
  void close() throws Exception;
}

File: core/rust/src/lang/protocol/iclose.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.iclose", name = "IClose")]
pub trait IClose {
    type Error;

    #[hara_method(value = "close", arity = 1)]
    fn close(&mut self) -> Result<(), Self::Error>;
}
